File: java/practicas/src/Ejercicio_001.java
// importar Scanner para poder esperar entrada de usuario
import java.util.Scanner;
// creation of principal class main()
class Ejercicio_001 {
    public static void main(String[] args) {
        // creation of the object Scanner to receive input
        Scanner input = new Scanner(System.in);

        // creation of variables to manage values
        boolean again = true;
        int bits, max_value;
        String sequence, confirmation, inputUser;
        
        // creation of warning messages
        String warning_size = "Your number is too big, the number must not pass 64";
        String warning_type = "Your input is incorrect, you have to pass a whole number";
        String warning_negative = "The number must be higher to 0";

        while (again) {
            // now we ask for the input user, that should be the number of bits and inside the while loop we check it
            while (true) {
                // use try to cacth an exception if the input's type is incorrect
                try {
                    // ask for the bits
                    System.out.println("Inserte el numero de bits: ");
                    inputUser = input.nextLine();
                    // checks that input user is not an empty value
                    if (inputUser.isEmpty()) {
                        System.out.println(warning_type);
                    } else {
                        bits = Integer.parseInt(inputUser);
                        // checks the number is higher than 0
                        if (bits <= 0) {
                            System.out.println(warning_negative);
                            continue;
                        }
                        // checks the number is less than 64
                        else if (bits > 64) {
                            System.out.println(warning_size);
                            continue;
                        }
                        break;
                    }
                }
                // capturing the exception for incorrect type
                catch (NumberFormatException e) {
                    System.out.println(warning_type);
                } finally {
                    System.out.println();
                }
            }
            // we calculate the max_value and the sequence here
            max_value = (1 << bits) - 1;
            sequence = "1".repeat(bits);

            // print the result of the value_max and the sequence of the bits
            String msj = String.format("The max value that can be represented with %d bits is: %d, and its sequence is: %s", bits, max_value, sequence);
            System.out.println(msj);
            System.out.println();

            // ask user to do or don't do another operation
            System.out.println("Quiere hacer otra operacion (No/Si)");
            confirmation = input.nextLine();

            // checks input and if not "si" main while loop ends
            if (!("si".equalsIgnoreCase(confirmation))) {
                again = false;
            }
        }
    }
}

File: python/practicas/ejercicio_001.py
again = True

# creation of warning messajes
warning_type = "The input that you enter is incorrect, must be a whole number"
warning_size = "Your number si too big, the number must not pass 64"
warning_negative = "The number must be higher to 0"

# creation of function that calculates the max_value and creates the sequence
def calcular_maximo(bits):
    # we will calculate the max_value and sequence of the bits
    max_value = 2**bits - 1
    sequence = "1" * bits 
    return max_value, sequence      

# creation of while loop of the program
while again:
    # while loop that controls input user is safe
    while True:
        try:
            # the user input that should be a whole number(type int)
            print()
            bits = int(input("Inserte los bits: "))
            # prevent input 0 or negatives numbers
            if bits <= 0:
                print(warning_negative)
                continue
            # prevent big numbers and memory
            elif bits > 64:
                print(warning_size)
                continue
            # if is correct input, breaks and go out of the loop
            break
        # if the input is not of type number 
        except ValueError:
            print(warning_type)
        finally:
            print()

    # call to the function that makes all the operation
    max_value, sequence = calcular_maximo(bits)

    # print the result of max_value and sequence of bits
    print(f"The max value that can be represent with {bits} bits is: {max_value}, and its sequence is: {sequence}")
    print()

    # the user input that should be a string
    confirmation = input("Quieres hacer otra operacion (No/Si): ")


    try:
        if not "si" == confirmation.lower():
            again = False
            break
    except ValueError:
        print(warning_type)
            


    
